### ISSUE TITLE
NVDA does not work in Avast 7 antivirus

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3241]
**Reported by jacobchawala on 2013-05-17 09:28**
NVDA does not work in Avast 7 antivirus. for that matter, this makes a person with visual impairment unable to access its buttons and other objects, menu bar, and other actions like update, delete, etc. after pressing enter to open Avast 7 on the desktop, nothing is responding even if you press alt, tab, application key, or any key. but NVDA does well in Microsoft security essentials antivirus. my wish is NVDA should also work well in Avast 7 and other related free antivirus.

**Comment by bhavyashah** [comment]
@jacobchawala Do you still experience such absolute inaccessibility with Avast using its latest stable release v. 17.6.2310 along with NVDA 2017.3? I have heard NVDA users discuss Avast as an anti virus software that is usable to a certain degree from an accessibility standpoint in more recent times, making me wonder if this more than four years old ticket still pertains. CC @Qchristensen (who has researched the topic of accessible anti virus software in the very recent past)

**Comment by Qchristensen** [comment]
TL;DR Avast is no better than it was four years ago, and I suspect that 99% of the issues lie with Avast itself.

So, the longer version: I downloaded and installed the latest version of Avast.  I just realised I never checked which version, but I think it's 17 now?  Anyway, here are my notes:

I used Windows 7, partly because that was the version that the original poster was likely using, and also Windows 10 users are more likely to be happy with the built-in Defender.

Running the downloaded installer stub, after confirming running it on the UAC, it gets straight to work.  It has a progress bar of sorts, but it's not standard and not accessible.

The installer proper seemed to have an odd tab order and some items appeared twice? Once on the install button, press spacebar specifically to install, enter doesn't work here.

After installing, the program (like most anti-virus programs) runs in the system tray.

Activating the desktop icon open the main window, but nothing is announced when the window loads.  Pressing tab, arrows and anything else I could think of didn't find anything useful.

The system tray does have some more options, however, it isn't easy to access either:
Press WINDOWS+B to move to the system tray and arrow to avast if needed.

The usual applications key or shift+f10 don't work to bring up the applications menu.

Press NVDA+numpad slash to move the mouse to the icon (NVDA+shift+m in laptop layout)
Press numpad asterisk twice (NVDA+] twice in laptop layout) to open the context menu (the first time it opens the context menu but at the same time it displays a pop-up advising that your system is protected which closes the context menu again).

From the context menu, you can update avast or disable shields for a set time.

Once you get sick of all that, you can't uninstall without a mouse user because the uninstall screen (which you can bring up from the add / remove programs options) is completely inaccessible.  I'd be inclined to mark this "NeedsExternalFix" unless anyone has any contacts at Avast who might be able to prioritise this from their end.

**Comment by Brian1Gaff** [comment]
I last communicated about three years back back they they told me that due 
to the way it protected itself, no access software could access it either 
and that disabling this kind of protection could be done but that left Avast 
itself open to attack, and at that time sent me a program which was supposed 
to uninstall it on exp. It did this, but I never followed it up as the 
impression I got was that tough, its the only way we can protect it.
 Obviously if anyone else wants to talk to them fine.
 I've lost all my emails from then now anyway.

As you rightly say, I'm not convinced many anti virus solutions are any 
better that microsofts and the occasional running of stand alone malware 
programs if you are suspicious.


Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
